### PR TITLE
[NPU] add batch_norm_op_npu and test

### PR DESCRIPTION
**Pull request by ronny1996**
### PR types

Others
### PR changes

OPs
### Describe

add batch_norm_op_npu and test

image

### CODE REVIEW

File: paddle/fluid/operators/batch_norm_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+#include "paddle/fluid/operators/batch_norm_op.h"
+
+#include "paddle/fluid/framework/fleet/ascend_wrapper.h"
+#include "paddle/fluid/framework/generator.h"
+#include "paddle/fluid/framework/op_registry.h"
+#include "paddle/fluid/framework/operator.h"
```
Comment (qili93) [inline paddle/fluid/operators/batch_norm_op_npu.cc]: 16-19行的头文件已经被batch_norm_op.h和npu_op_runner.h包含，可以不需要？

File: paddle/fluid/operators/batch_norm_op_npu.cc [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+#include "paddle/fluid/operators/fill_constant_op.h"
```
Comment (qili93) [inline paddle/fluid/operators/batch_norm_op_npu.cc]: fill_constant_op.h也可以不需要？

File: python/paddle/fluid/tests/unittests/npu/test_batch_norm_op_npu.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+    def test_check_output(self):
+        place = core.NPUPlace(0)
+        for data_format in self.data_formats:
+            self.check_with_place(place, data_format, self.dtype, [2, 3, 4, 5])
+            self.check_with_place(place, data_format, self.dtype, [2, 3])
+
+    def init_kernel_type(self):
+        pass
+
+
```
Comment (qili93) [inline python/paddle/fluid/tests/unittests/npu/test_batch_norm_op_npu.py:426]: 最好也参考 test_batch_norm_op.py把里面所有的test case加上，不能支持的就注释一下说明原因，尽量保证单测的coverage。

**Review by qili93 — CHANGES_REQUESTED** [review]

File: paddle/fluid/memory/allocation/allocator_facade.cc [hunk 4]
```diff
@@ -160,6 +161,16 @@ class AllocatorFacadePrivate {
...
+#endif
+#if defined(PADDLE_WITH_ASCEND_CL)
+    system_allocators_[platform::NPUPinnedPlace()] =
+        std::make_shared<paddle::memory::allocation::NPUPinnedAllocator>();
+    int device_count = platform::GetNPUDeviceCount();
+    for (int i = 0; i < device_count; ++i) {
+      platform::NPUPlace p(i);
+      system_allocators_[p] =
+          std::make_shared<paddle::memory::allocation::NPUAllocator>(p);
+    }
```
Comment (zhiqiu) [inline paddle/fluid/memory/allocation/allocator_facade.cc]: Why add pinned allocator here? Is it needed in batch_norm op?

Comment (ronny1996) [reply paddle/fluid/memory/allocation/allocator_facade.cc]: It is no longer needed and has been removed

**Review by qili93 — DISMISSED** [review]
LGTM

**Review by zhiqiu — APPROVED** [review]
LGTM for ShareDataWith

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**